pkg/cloud/libvirt/actuators/machine/actuator.go

```diff
@@ -284,6 +284,12 @@ func (a *Actuator) createVolumeAndDomain(ctx context.Context, machine *machinev1
 		if err := client.DeleteVolume(domainName); err != nil {
 			glog.Errorf("Error cleaning up volume: %v", err)
 		}
+		if err := client.DeleteVolume(cloudInitVolumeName(domainName)); err != nil {
+			glog.Errorf("Error cleaning up cloud-init volume: %v", err)
```

Comment:
throw the error only when `err != libvirtclient.ErrVolumeNotFound ` ?

Reply:
fixed!